src/OmniSharp/Api/Navigation/OmnisharpController.NavigateUpDown.cs

```diff
@@ -0,0 +1,74 @@
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.AspNet.Mvc;
+using OmniSharp.Models;
+using System;
+
+namespace OmniSharp
+{
+    public partial class OmnisharpController
+    {
+        private Func<Request, FileMemberElement, GotoDefinitionResponse, bool> NavigateUpFunc = (request, node, qf) =>
+        {
+            bool retVal = false;
+            if (node.Location.Line < request.Line || (node.Location.Line == request.Line && node.Location.Column < request.Column))
+            {
+                qf.Line = node.Location.Line;
+                qf.Column = node.Location.Column;
+                retVal = true;
+            }
+            return retVal;
+        };
+
+        private Func<Request, FileMemberElement, GotoDefinitionResponse, bool> NavigateDownFunc = (request, node, qf) =>
+        {
+            bool retVal = false;
+            if (node.Location.Line > request.Line || (node.Location.Line == request.Line && node.Location.Column > request.Column))
+            {
+                qf.Line = node.Location.Line;
+                qf.Column = node.Location.Column;
+                retVal = true;
+            }
+            return retVal;
+        };
+
+        private async Task<IActionResult> Navigate(Request request, bool toBreak, Func<Request, FileMemberElement, GotoDefinitionResponse, bool> navigateFunc)
+        {
+            var stack = new List<FileMemberElement>(await StructureComputer.Compute(_workspace.GetDocuments(request.FileName)));
+            var ret = new GotoDefinitionResponse();
```

Comment:
Shouldn't really be returning a GotoDefinitionResponse here (although we need an object with the same signature)